File: leetcode/2938-separate-black-and-white-balls/minimumSteps.ts
function minimumSteps(s: string): number {
  let steps: number = 0;
  let blackCount: number = 0;
  for (const color of s) {
    if (color === "1") {
      blackCount++;
    } else {
      steps += blackCount;
    }
  }

  return steps;
}

export default minimumSteps;

File: leetcode/2938-separate-black-and-white-balls/minimumSteps.rs
use crate::Solution;

impl Solution {
    pub fn minimum_steps(s: String) -> i64 {
        let mut steps: i64 = 0;
        let mut black_count: i64 = 0;

        for color in s.chars() {
            if color == '1' {
                black_count += 1;
            } else {
                steps = steps + black_count
            }
        }

        return steps;
    }
}
